.github/scripts/trymerge.py

```diff
@@ -652,6 +653,25 @@ def merge_into(self, repo: GitRepo, *, force: bool = False, dry_run: bool = Fals
 
         repo.push(self.default_branch(), dry_run)
 
+    def merge_on_green(self, repo: GitRepo, org: str, project: str, pr_num: int, *, dry_run: bool = False) -> None:
+        start_time = time.time()
+
+        while True:
+            current_time = time.time()
+            elapsed_time = current_time - start_time
+
+            if(elapsed_time > 355 * 60):
+                msg = 'Took too long to merge. Please try again later'
+                gh_post_comment(org, project, pr_num, msg, dry_run=dry_run)
+                gh_add_labels(org, project, pr_num, ["land-failed"])
+                raise RuntimeError(msg)
+
+            try:
+                self.merge_into(repo, dry_run=dry_run)
```

Comment:
Wouldn't this still post confusing comments on the PR that merge failed every 60s? 

You might need to change some logic in the other parts of the script if ON_GREEN is true

Reply:
I don't believe so. The merge_into method doesn't contain anything that would post a comment. It only posts a comment when they catch the exception thrown from within it like on line L856.